nose_parameterized/parameterized.py

```diff
@@ -211,11 +211,53 @@ def expand(cls, input, testcase_func_name=None):
             >>>
             """
 
-        def parameterized_expand_wrapper(f):
+        def doc_expansion(f, args, kwargs, is_method):
+            offset = 1 if is_method else 0
+            vns = inspect.getargspec(f)[0][offset:]
+            descs = ["{0} = {1}".format(n, repr(v)) for n, v in zip(vns, args)]
+            descs.extend(["{0} = {1}".format(k, repr(v)) for k, v in kwargs.items()])
+
+            # Make up for the shortfall by setting the rest to None
+            descs.extend(["{0} = {1}".format(n, "None") for n in vns[len(args) + len(kwargs.keys()):]])
+
+            if f.__doc__ is not None:
+                # The documentation might be a multiline string, so split it
+                # and just work with the first string, ignoring the period
+                # at the end if there is one.
+                split_doc = f.__doc__.split("\n")
+                first = split_doc[0]
+                append = ""
+                if first[-1] == ".":
+                    append = "."
+                    first = first[:-1]
+
+                first = first + " [with {0}]".format(", ".join(descs)) + append
+                return "\n".join([first] + split_doc[1:])
+            else:
+                return None
```

Comment:
Hmmm, I'm not sure what the best thing to do here is.

I know that test runners which do look at the function's `__doc__`  will default to showing the function name instead of the documentation if `__doc__` is set to None. I suspect those same test runners don't treat `None` and `""` in the same way, which could lead to blank-line-tests.